### PR TITLE
fix: make license parsing and protocol more resilient

### PR DESCRIPTION
**Pull request by ibeckermayer**
A user encountered

```
ConnectorError(Error { context: "decode SERVER_PLATFORM_CHALLENGE", kind: Pdu(Error { context: "<ironrdp_pdu::rdp::server_license::BlobHeader as ironrdp_pdu::PduDecode>::decode::{{c
losure}}", kind: InvalidMessage { field: "blobType", reason: "invalid blob type" }, source: None }), source: None })
```

This appears to be a bug in the parsing code for `SERVER_PLATFORM_CHALLENGE` which contains a Licensing Binary BLOB[<12>](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpele/1040af38-c733-4fb3-acd1-8db8cc979eda#Appendix_A_12) with the following footnote:

> The wBlobType field is unused in EncryptedPlatformChallenge (part of SERVER_PLATFORM_CHALLENGE PDU), which is sent by the server to the client.

suggesting that the blob type may not be one of the enumerated expected values.

This PR re-introduces resilient parsing for this field. 

~Putting this in draft mode pending an in-field test by the user.~

**Update:**

After attempting to connect over xrdp, I discovered a larger refactor should be done to improve the resiliency of the licensing protocol. This has also now been tested against the standard windows rdp server.

### CODE REVIEW

**Review by CBenoit — APPROVED** [review]
Thank you! Looks good to me! Let’s merge once the user confirms everything is good.

**Comment by ibeckermayer** [comment]
Substantial changes have been made since approval see the `**Update**` section in the description.

@CBenoit @pacmancoder

**Comment by pacmancoder** [comment]
@ibeckermayer Benoit is on vacation, I'll take a look at it today👍

**Review by pacmancoder — COMMENTED** [review]
I have only a few minor comments to resolve before we merge, but overall a great job 👍

**Comment by CBenoit** [comment]
Thank you for this @ibeckermayer 
I adjusted the name of the PR (for when it will be squashed). Otherwise I leave the rest to @pacmancoder :raised_hands:

File: crates/ironrdp-pdu/src/rdp/headers.rs
Comment (CBenoit) [reply crates/ironrdp-pdu/src/rdp/headers.rs:325]: I mostly agree! Here is an alternative:

```rust
struct ShareDataPduRaw<'a> {
    kind: ShareDataPduType,
    data: Cow<'a, [u8]>,
}

impl ShareDataPdu<'_> {
     fn parse(&self) -> Result<ShareDataPdu, …> { … }
}
```

With `ShareDataPdu` only including supported PDUs, and allowing graceful handling both known and unknown PDU types.

That being said, if the plan is to support all the different types soon enough, it’s probably fine to just add the new variants now instead. I don’t think the protocol will change that much, and if it does, it will likely be controlled by compatibly flags during the handshake.

File: crates/ironrdp-fuzzing/src/oracles/mod.rs
Comment (ibeckermayer) [reply crates/ironrdp-fuzzing/src/oracles/mod.rs:42]: derp 🤪

File: crates/ironrdp-pdu/src/rdp/capability_sets.rs
Comment (ibeckermayer) [reply crates/ironrdp-pdu/src/rdp/capability_sets.rs:271]: My understanding is that the groupings here are

- [mandatory](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/c21e61b5-82a5-4e95-a660-fa95328c2712)
- [optional](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/c7bf0bde-50e3-48e6-bd41-a296b8ff5b35)
- other: from specs besides [MS-RDPBCGR]

File: crates/ironrdp-pdu/src/rdp/headers.rs
Comment (ibeckermayer) [reply crates/ironrdp-pdu/src/rdp/headers.rs:325]: I'm going to punt on this one for now.

**Review by pacmancoder — APPROVED** [review]
Looks good to me!

**Pull request merged**
